### PR TITLE
doc: pydocstyle ruff rule

### PR DESCRIPTION
**Pull request by moe-ad**
This PR is related to #296. The codebase will be modified to conform with pydocstyle guidelines.

### CODE REVIEW

File: pyproject.toml
Comment (jorgepiloto) [inline pyproject.toml:1]: How are we excluding the auto-generated files from being linted?

**Review by jorgepiloto — APPROVED** [review]
Amazing work, @moe-ad. Just left one question.

File: pyproject.toml
Comment (moe-ad) [reply pyproject.toml:1]: There is a [regular expression](https://github.com/ansys/pydpf-core/blob/0d7bba792e11198cce47bcaabd7075b648be111c/.pre-commit-config.yaml#L1) in pre-commit-config.yaml that matching directories we are excluding.

**Comment by moe-ad** [comment]
Thank you @jorgepiloto... 🕺

File: pyproject.toml
Comment (jorgepiloto) [reply pyproject.toml:1]: Oh! Thanks for clarifying. Then, let's get this merged.

**Pull request merged**